server/src/main/java/keywhiz/service/permissions/OwnershipPermissionCheck.java

```diff
@@ -0,0 +1,76 @@
+package keywhiz.service.permissions;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Inject;
+import java.util.Set;
+import keywhiz.api.model.Client;
+import keywhiz.api.model.Group;
+import keywhiz.api.model.Secret;
+import keywhiz.service.daos.AclDAO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OwnershipPermissionCheck implements PermissionCheck{
+  private static final Logger logger = LoggerFactory.getLogger(
+      OwnershipPermissionCheck.class);
+  private static final String successMetricName = MetricRegistry.name(OwnershipPermissionCheck.class, "success", "histogram");
+  private static final String failureMetricName = MetricRegistry.name(OwnershipPermissionCheck.class, "failure", "histogram");
+
+  private MetricRegistry metricRegistry;
+
+  private final AclDAO aclDAO;
+
+  @Inject
+  public OwnershipPermissionCheck(MetricRegistry metricRegistry, AclDAO.AclDAOFactory aclDAOFactory) {
+    this.metricRegistry = metricRegistry;
+    this.aclDAO = aclDAOFactory.readwrite();
+  }
+
+  public boolean isAllowed(Object source, String action, Object target) {
+    boolean hasPermission = false;
+
+    if (isClient(source) && isSecret(target)) {
+      Set<Group> clientGroups = aclDAO.getGroupsFor((Client) source);
+      String secretOwner =((Secret) target).getOwner();
```

Comment:
We should really get Checkstyle going for this project, but in the meantime make sure you always put a space on both sides of an equal sign.

Reply:
there is. just realized that the style guide warned me about the if statement simplification for the help functions. but not the spacing